src/connector/src/sink/doris_starrocks_connector.rs

```diff
@@ -337,3 +378,234 @@ impl InserterInner {
         self.wait_handle().await
     }
 }
+
+pub struct MetaRequestSender {
+    client: Client,
+    request: Request,
+    fe_host: String,
+}
+
+impl MetaRequestSender {
+    pub fn new(client: Client, request: Request, fe_host: String) -> Self {
+        Self {
+            client,
+            request,
+            fe_host,
+        }
+    }
+
+    /// Send the request and handle redirection if any.
+    /// The reason we handle the redirection manually is that if we let `reqwest` handle the redirection
+    /// automatically, it will remove sensitive headers (such as Authorization) during the redirection,
+    /// and there's no way to prevent this behavior.
+    ///
+    /// Another interesting point is that some of the `StarRocks` transactional APIs will respond directly from FE.
+    /// For example, the request to `/api/transaction/commit` endpoint does not seem to redirect to BE.
+    pub async fn send(self) -> Result<Bytes> {
+        let request = self.request;
+        let mut request_for_redirection = request.try_clone().ok_or_else(|| {
+            SinkError::DorisStarrocksConnect(anyhow::anyhow!("Can't clone request"))
+        })?;
+
+        let resp = self
+            .client
+            .execute(request)
+            .await
+            .map_err(|err| SinkError::DorisStarrocksConnect(err.into()))?;
+
+        let be_url = try_get_be_url(&resp, self.fe_host)?;
```

Comment:
Looks like `send` will be called multiple times, is it? If so, please keep the `be_url` and reuse it as much as possible.

Reply:
same reason as described above